rfcs/proposed/task_group_dynamic_dependencies/README.md

```diff
@@ -0,0 +1,365 @@
+returns, based on application logic, the tasks that must complete 
+before the new work can start.
+
+    void add_another_task(tbb::task_group& tg, int work_id) {
+        tbb::task_handle new_task = tg.defer([=] { do_work(work_id); });
+
+        for (tbb::task_handle& p : users::find_predecessors(work_id)) {
+            new_task.add_predecessor(p);
+        }
+
+        tg.run(new_task);
+    }
+
+Again, the graph, as shown below, is simple. However, now we do not 
+know the completion status of the predecessors. Therefore, 
+for ease-of-use, a `task_handle` should be usable as a dependency 
+regardless of state of the task it represents. Any predecessor 
+that is already completed when it is added as a predecessor will 
+not delay the start of the dependent task. Otherwise, end-users 
+will need to track these states explicitly.
+
+<img src="unknown_states.png" width=400>
+
+#### And example of recursive decomposition
+
+This example is a version of merge-sort (with many of the details left out). 
+Assume that there is an initial task that executes the function shown 
+below as its body, and the function implements that task, and also serves 
+as the body for the recursively decomposed pieces. The beginning and
+end of the sequence are represented by `b` and `e`, and much of the
+(unimportant) details of the implementation of merge-sort is hidden 
+behind the functions `users::do_serial_sort`, `users::create_left_range`,
+`users::create_right_range`, and `users::do_merge`.  
+
+    template<typename T>
+    void merge_sort(tbb::task_group& tg, T b, T e) {
+        if (users::range_is_too_small(b, e)) {
+            // base-case when range is small
+            users::do_serial_sort(b, e);
+        } else {
+            // calculate left and right ranges
+            T lb, le, rb, re;
+            users::create_left_range(lb, le, b, e);
+            users::create_right_range(rb, re, b, e);
+
+            // create the three tasks
+            tbb::task_handle sortleft =
+                tg.defer([lb, le, &tg] {
+                    merge_sort(tg, lb, le);
+                });
+            tbb::task_handle sortright =
+                tg.defer([rb, re, &tg] {
+                    merge_sort(tg, rb, re);
+                });
+            tbb::task_handle merge =
+            tg.defer([rb, re, &tg] {
+                    users::do_merge(tg, lb, le, rb, re);
+                });
+
+            // add predecessors for new merge task
+            merge.add_predecessors(sortleft, sortright);
+
+            // insert new subgraph between currently executing
+            // task and its successors
+            tbb::transfer_successors_to(merge);
+
+            tg.run(sortleft);
+            tg.run(sortright);
+            tg.run(merge);
+        }
+    }
+
+The task tree for this example matches the one shown earlier for merge-sort. 
+
+## Open Questions in Design
+
+Some open questions that remain:
```

Comment:
Should there be more implementation-level questions added, such as feasibility of implementing the proposed design, performance implications, etc.?